client/src/components/advancedSearch/DateRangeFilter.tsx

```diff
@@ -183,12 +203,16 @@ export const deserialize = ({ queryKey }, query, key) => {
   const filterValue = {}
 
   if (query[startKey]) {
-    toQueryValue[startKey] = query[startKey]
+    const startVal = query[startKey]
+    toQueryValue[startKey] = startVal
     const lastValues = [LAST_DAY, LAST_WEEK, LAST_MONTH]
-    if (lastValues.indexOf(+query[startKey]) !== -1) {
-      filterValue.relative = query[startKey]
+    if (lastValues.indexOf(+startVal) !== -1) {
+      filterValue.relative = startVal
+    } else if (!isNaN(parseInt(startVal))) {
+      filterValue.relative = LAST_X_DAYS
+      filterValue.days = parseInt(startVal)
     } else {
-      filterValue.start = moment(query[startKey]).format(DATE_FORMAT)
+      filterValue.start = moment(startVal).format(DATE_FORMAT)
       if (query[endKey]) {
```

Comment:
Suggest to change to `if (Object.hasOwn(query, endKey)) {` so `0` is also valid.
And make similar changes to the `if` block below:
```javascript
  if (Object.hasOwn(query, endKey)) {
    filterValue.end = moment(query[endKey]).format(DATE_FORMAT)
    toQueryValue[endKey] = query[endKey]
    if (!Object.hasOwn(query, startKey)) {
      filterValue.relative = BEFORE
    }
  }
```